lib/datadog/profiling/agent_settings_resolver.rb

```diff
@@ -0,0 +1,335 @@
+# frozen_string_literal: true
+
+require 'uri'
+
+require_relative '../core/configuration/settings'
+require_relative '../core/configuration/agent_settings_resolver_common'
+require_relative '../tracing/configuration/ext'
+require_relative '../../ddtrace/transport/ext'
+
+module Datadog
+  module Profiling
+    # This class unifies all the different ways that users can configure how we talk to the agent.
```

Comment:
TODO: Add comment explaining why this is here and the same as tracing's resolver.